### DISCUSSION TITLE
Override Icon component base style (height and width)

### DISCUSSION THREAD

**Discussion by muradsofi** [chakra-ui/chakra-ui#4597] [Q&A]
We want to change <Icon/> component base height and width. Currently, Icon component height and width is 1em but we want to update this to auto.  Is it possible to change Icon component style in theme customization?

**Comment by chasinhues** [comment]
It looks like `Icon` height and width are hard-set in the component, not the theme.

https://github.com/chakra-ui/chakra-ui/blob/d1da680a2c29cf183a4f69bc1c5cc2aed0629345/packages/icon/src/icon.tsx#L49

So, you won't be able to do this in the theme. You can create your own Icon component that uses Chakra's Icon component, and explicitly sets height and width to auto:

https://codesandbox.io/s/clever-brown-qe94s?file=/src/AutosizeIcon.tsx

Note, you have to forward the ref to your Icon component because Chakra's `Icon` uses the `as` prop.

**Reply by muradsofi** [reply]
Yeah, I know that but why Chakra did not add this Icon component to the theme? :/
